ingenialink/ethernet/dictionary.py

```diff
@@ -0,0 +1,68 @@
+from ingenialink.dictionary import Dictionary, AttrRegDict
+from ingenialink.constants import SINGLE_AXIS_MINIMUM_SUBNODES
+from ingenialink.ethernet.register import EthernetRegister
+
+import ingenialogger
+
+logger = ingenialogger.get_logger(__name__)
+
+
+class AttrRegEthDict(AttrRegDict):
+    ADDR = 'address'
+
+
+class EthernetDictionary(Dictionary):
+    """Contains all registers and information of a Ethernet dictionary.
+
+    Args:
+        dictionary_path (str): Path to the Ingenia dictionary.
+
+    """
+
+    def __init__(self, dictionary_path):
+        super().__init__(dictionary_path)
+        self.version = '1'
+        self.subnodes = SINGLE_AXIS_MINIMUM_SUBNODES
+
+        self.read_dictionary()
+
+    def _read_register(self, register):
+        """Reads a register from the dictionary and creates a Register instance.
+
+        Args:
+            register (Element): Register instance from the dictionary.
+
+        """
+        try:
+            current_read_register = super()._read_register(register)
+
+            current_read_register[AttrRegEthDict.ADDR] = int(register.attrib['address'], 16)
+
+            return current_read_register
+
+        except KeyError as ke:
```

Comment:
This try-except changes the function behavior. The parent function can't return None, but this function does.